LaTeX/LaTeX.sublime-syntax

```diff
@@ -628,6 +629,41 @@ contexts:
           pop: true
         - include: main
 
+  footnote:
+    - match: |-
+        (?x)
+        ((\\)footnote(?:text)?)\s*
+        ((\[)
+            \s*\d*\s*
+        (\]))?\s*
+        (\{)
+      captures:
+        1: support.function.footnote.latex
+        2: punctuation.definition.backslash.latex
+        3: meta.group.bracket.latex
+        4: punctuation.definition.group.bracket.begin.latex
+        5: punctuation.definition.group.bracket.end.latex
+        6: meta.group.brace.latex punctuation.definition.group.brace.begin.latex
+      push:
+        - meta_scope: meta.function.footnote.latex
+        - meta_content_scope: meta.group.brace.latex markup.italic.footnote.latex
+        - match: '\}'
+          scope: meta.group.brace.latex punctuation.definition.group.brace.end.latex
+          pop: true
+        - include: main
+    - match: |-
+        (?x)
+        ((\\)footnotemark)\s*
+        ((\[)
```

Comment:
You could use pattern `optional-arguments` to match the optional arguments.

Reply:
Yes that's better